### PR TITLE
fix sending proof

### PR DESCRIPTION
**Pull request by MixailE**

### CODE REVIEW

File: src/components/Founder.tsx [hunk 1]
```diff
@@ -23,23 +23,20 @@ const container = classnames(
   alignItems('items-center'),
   space('space-y-2'),
   padding('px-8'),
-  gap('gap-2')
+  gap('gap-2'),
+  backgroundColor('!bg-black')
```
Comment (T-Damer) [inline src/components/Founder.tsx]: We don't need important here

Comment (T-Damer) [reply src/components/Founder.tsx]: Also, should it be black? Not transparent?

File: src/components/Founder.tsx [hunk 2]
```diff
@@ -57,10 +54,26 @@ export default function Founder() {
...
   }, [])
 
-  return (
-    <div className={container}>
-      <KetlLogo />
-      <Description />
-    </div>
-  )
+  useEffect(() => {
+    if (data) {
```
Comment (T-Damer) [inline src/components/Founder.tsx]: `if (!data) return`

File: src/components/Founder.tsx [hunk 3]
```diff
@@ -57,10 +52,26 @@ export default function Founder() {
...
+    })
+    .catch((e) => {
+      postWebViewMessage({
+        data: {
+          message: `Can't generate valid proof with this token`,
+          e: JSON.stringify(e, Object.getOwnPropertyNames(e)),
+        },
+        type: Messages.GetFounderProofError,
+      })
+    })
```
Comment (smyzin) [inline src/components/Founder.tsx]: It's look strange, specially with `void`. Can we move it to `helpers` or even better, use `useCallback` here?

**Review by smyzin — APPROVED** [review]

File: src/components/Founder.tsx [hunk 4]
```diff
@@ -57,10 +58,29 @@ export default function Founder() {
...
+        postWebViewMessage({
+          data: proof,
+          type: Messages.GetFounderProof,
+        })
+      })
+      .catch((e) =>
+        postWebViewMessage({
+          data: {
+            message: `Can't generate valid proof with this token`,
+            e: JSON.stringify(e, Object.getOwnPropertyNames(e)),
```
Comment (T-Damer) [inline src/components/Founder.tsx]: I hope it won't fail

**Review by T-Damer — APPROVED** [review]

File: src/helpers/checkIfProofUsedBefore.ts [hunk 5]
```diff
@@ -1,10 +1,5 @@
-import AllowListType from 'models/AllowListType'
-import getAllowList from 'helpers/getAllowList'
+import getKetlAttestationContract from './getKetlAttestationContract'
```
Comment (T-Damer) [inline src/helpers/checkIfProofUsedBefore.ts]: Should we use absolute import?

File: src/helpers/createProof.ts [hunk 6]
```diff
@@ -1,35 +1,64 @@
...
+  }
+
+  const proof = await snarkjs.groth16.fullProve(
+    inputs,
+    'zk/AttestationChecker.wasm',
+    'zk/AttestationChecker_final.zkey'
+  )
+
+  if (await checkIfProofUsedBefore(proof.publicSignals[3])) {
+    return Promise.reject(new Error(`This token has already been used!`))
```
Comment (T-Damer) [inline src/helpers/createProof.ts]: Let's use simple braces, not ``

File: src/helpers/requestSignature.ts [hunk 7]
```diff
@@ -8,12 +9,35 @@ export enum VerificationType {
   email = 'email-unique',
   twitter = 'twitter',
   balance = 'balance-unique',
+  token = 'token',
 }
 
-export async function requestSignature(type: VerificationType, params: object) {
+export enum VerificationId {
```
Comment (T-Damer) [inline src/helpers/requestSignature.ts]: I think this should be in `models`

File: src/hooks/useProof.ts [hunk 8]
```diff
@@ -0,0 +1,53 @@
+import { useCallback, useState } from 'preact/hooks'
+import CreateProofParams from 'models/CreateProofParams'
+import Messages from 'models/Messages'
+import generateProof from 'helpers/createProof'
+import postWebViewMessage from 'helpers/postWebViewMessage'
+
+function isValidProofMessage(params: object): params is CreateProofParams {
```
Comment (T-Damer) [inline src/hooks/useProof.ts]: Let's extract this one into `helpers`

File: src/hooks/useMessageHandler.ts [hunk 9]
```diff
@@ -0,0 +1,47 @@
+import { useEffect, useState } from 'preact/hooks'
+import Message from 'models/Message'
+import isDataInMessage from 'helpers/isDataInMessage'
+
+function isMessage(data: object): data is Message {
```
Comment (T-Damer) [inline src/hooks/useMessageHandler.ts]: Should we merge `isDataInMessage` with this helper?

Comment (T-Damer) [reply src/hooks/useMessageHandler.ts]: At least let's extract it

File: src/helpers/createProof.ts [hunk 10]
```diff
@@ -1,35 +1,64 @@
...
+}
+
+async function getHashes(id: number) {
+  const response = await fetch(`/trees/${id}.json`)
+
+  return response.json()
+}
+
+export default async function createFounderProof(params: CreateProofParams) {
+  const hashes = await getHashes(params.id)
```
Comment (smyzin) [inline src/helpers/createProof.ts]: Should it be wrapped with `try/catch`, wdyt?

**Review by smyzin — APPROVED** [review]

File: src/helpers/createProof.ts (hunk 10 above)
Comment (MixailE) [reply src/helpers/createProof.ts]: We'll catch the error inside `useProof`

**Comment by MixailE** [comment]
[Update OBSS in ketl-zk-frontend](https://trello.com/c/GGDe7tKe/2827-update-obss-in-ketl-zk-frontend)

**Pull request merged**